bin/publish-npm-modules.sh

```diff
@@ -63,13 +79,41 @@ mkdir -p "$SOCKET_HOME"
 export SOCKET_HOME
 export PREFIX
 
-if (( do_global_link && !dry_run )); then
-  echo >&2 "warning: '--link' implies '--dry-run' too"
-  dry_run=1
+if (( !only_top_level ))  && (( !no_rebuild )) ; then
+  "$root/bin/install.sh" || exit $?
 fi
 
-if (( !only_top_level )); then
-  "$root/bin/install.sh" || exit $?
+declare ABORT_ERRORS=0
+
+
+# Confirm that build SSC matches current commit
+if command -v ssc >/dev/null 2>&1; then
+  REPO_VERSION="$(cat "$root/VERSION.txt") ($(git rev-parse --short=8 HEAD))"
+  BUILD_VERSION=$("$SOCKET_HOME"/bin/ssc --version)
+
+  if [[ "$REPO_VERSION" != "$BUILD_VERSION" ]]; then
+    echo "Repo $REPO_VERSION and $SOCKET_HOME/bin/ssc $BUILD_VERSION don't match."
+    ABORT_ERRORS=1
+  fi
+fi
+
+for abi in $(android_supported_abis); do
+  lib_path="$SOCKET_HOME/lib/$abi-android"
+  if [[ ! -f "$lib_path/libuv.a" ]]; then
+    ABORT_ERRORS=1
+    echo "$lib_path/libuv.a missing - check build process."
+  fi
+  
+  if [[ ! -f "$lib_path/libsocket-runtime.a" ]]; then
+    ABORT_ERRORS=1
+    echo "$lib_path/libsocket-runtime.a missing - check build process."
+  fi
+  export ABORT_ERRORS
+done
+
+if (( ABORT_ERRORS )); then
+  echo "Refusing to publish due to missing static libs."
```

Comment:
I think

```sh
echo "Refusing to publish due to errors."
```

makes more sense because it looks like other things could cause `ABORT_ERRORS` to be `1`